src/utils.rs

```diff
@@ -2,14 +2,52 @@ use std::{error::Error, fmt::Display};
 
 use napi::{bindgen_prelude::BigInt, Status};
 
+pub enum ErrorType {
+    ArgumentError,
+    AuthenticationError,
+    BusyConnectionError, // TODO: Add suport for fields of this error
+    DriverError,
+    DriverInternalError,
+    NoHostAvailableError, // TODO: Add suport for fields of this error
+    NotSupportedError,
+    OperationTimedOutError, // TODO: Add suport for fields of this error
+    ResponseError,          // TODO: Add suport for fields of this error
+    Error,
+    RangeError,
+    ReferenceError,
+    SyntaxError,
+    TypeError,
+}
+
```

Comment:
Have you added an issue regarding those TODOs?

Reply:
Yes, issue #119